### PR TITLE
Setup script

### PR DESCRIPTION
**Pull request by netshade**
This PR adds a setup script and some adjustments to allow for an easier introduction to the project. It makes the following changes:

1. Documents the expected Ruby version in `.ruby-version` ( coinciding with a compatible file with `rbenv` )
1. Documents the expected Rust nightly in `REQUIRED_RUST_NIGHTLY`
1. Uses a `Gemfile` to include the `foreman` and `sass` dependencies as expected with Ruby
1. Adds a `scripts/setup.macos` file to perform the steps outlined in the `README` to bring an environment up

If this PR is desirable, would suggest that:

1. It be tested on Linux, and if it works reliably there, the setup script be renamed to `scripts/setup`
1. The `README` be changed to reflect that an environment shall be setup by running `scripts/setup`, and that the expectation should be that any environment shall be made correct for running the application by running `scripts/setup`.
1. The TravisCI setup should likely use `scripts/setup` as a way to bring the environment up

### CODE REVIEW

**Comment by netshade** [comment]
Would note that the items I put as desirable I consider to be tasks that I should do before this PR got merged, but that it was worth checking if this was even desirable to the project before going forward.

**Comment by yipdw** [comment]
The setup scripts look okay to me; I'll give it a shot in a bit.  `scripts/setup.macos` _looks_ like it should apply to Linux as well, but I'll know once I get to run it.

**Comment by netshade** [comment]
I adjusted the script name to `scripts/setup`, setup the Travis CI config to use it ( and also copied back some expectations that were in the Travis CI config into `setup` ), and adjusted the `README` such that the new setup expectations were communicated.

**Comment by barzamin** [comment]
I'm going to test this out on a fresh clone, and then, assuming it works (@yipdw how did your test go, by the way), this looks good to go!

**Comment by barzamin** [comment]
Oh, one note about the setup script - it would perhaps be nice if it echoed some of the commands it was running (for debugging purposes, and illustration that it's not performing magic, especially when some of these commands sit for a while without producing output).

File: scripts/setup [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# Diesel CLI install
+which diesel 2>/dev/null 1>&2 || cargo install diesel_cli --no-default-features --features="postgres"
+which cargo-watch 2>/dev/null 1>&2 || cargo install cargo-watch
+
+# Database creation and migration
+diesel database setup --migration-dir=migrations
+
+# Cargo build
+if [ -z "${TRAVIS_BRANCH}" ]; then
+  cargo build
```
Comment (barzamin) [inline scripts/setup]: I'm not entirely convinced doing this in the setup script is necessary or desirable, considering that `foreman start` will do the same thing by running `cargo run`.

Comment (netshade) [reply scripts/setup]: Fair - I struggled with the right thing to do here, whether or not `setup` would preflight the entire process, or just bring the environment into a ready state. It does seem like an odd wart here tho, so I'll just remove.

**Comment by netshade** [comment]
@barzamin I added some better messages about what was taking place before each step, as well as a note about `set -x` if the user wants more information about what's taking place.

**Comment by barzamin** [comment]
great, thanks! :purple_heart:

**Comment by yipdw** [comment]
Whoops, sorry about the delay.  `scripts/setup` works for me; the only nitpick I have (and it's a small one) is that I'd like Bundler to install to a local path to avoid needing sudo privileges, e.g.

```
bundle install --path=`pwd`/vendor
```

  However, there are other ways to achieve that (rbenv, rvm, ...) and so I don't think it needs to be changed just yet.

**Pull request merged**